event/webhook/webhook.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2018 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/tsuru/tsuru/api/shutdown"
+	tsuruErrors "github.com/tsuru/tsuru/errors"
+	"github.com/tsuru/tsuru/event"
+	"github.com/tsuru/tsuru/log"
+	tsuruNet "github.com/tsuru/tsuru/net"
+	"github.com/tsuru/tsuru/storage"
+	eventTypes "github.com/tsuru/tsuru/types/event"
+	"github.com/tsuru/tsuru/validation"
+)
+
+var (
+	_ eventTypes.WebHookService = &webHookService{}
+
+	chanBufferSize   = 1000
+	defaultUserAgent = "tsuru-webhook-client/1.0"
+
+	webhooksLatency = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name: "tsuru_webhooks_latency_seconds",
+		Help: "The latency for webhooks requests in seconds",
+	})
+
+	webhooksQueue = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "tsuru_webhooks_event_queue_current",
+		Help: "The current number of queued events waiting for webhooks processing",
+	})
+
+	webhooksTotal = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "tsuru_webhooks_calls_total",
+		Help: "The total number of webhooks calls",
+	})
+
+	webhooksError = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "tsuru_webhooks_calls_error",
+		Help: "The total number of webhooks calls with error",
+	})
+)
+
+func init() {
+	prometheus.MustRegister(webhooksLatency, webhooksQueue, webhooksTotal, webhooksError)
+}
+
+func WebHookService() (eventTypes.WebHookService, error) {
+	dbDriver, err := storage.GetCurrentDbDriver()
+	if err != nil {
+		dbDriver, err = storage.GetDefaultDbDriver()
+		if err != nil {
+			return nil, err
+		}
+	}
+	s := &webHookService{
+		storage: dbDriver.WebHookStorage,
+		evtCh:   make(chan string, chanBufferSize),
+		quitCh:  make(chan struct{}),
+		doneCh:  make(chan struct{}),
+	}
+	go s.run()
+	shutdown.Register(s)
+	return s, nil
+}
+
+type webHookService struct {
```

Comment:
You're mixing `Webhook` and `WebHook` cases. We should choose one of them

Reply:
Going with `Webhook`